src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,116 @@
+package duke.storage;
+
+import duke.task.*;
+import duke.exception.DukeException;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The Storage class handles the reading and writing of data from the hard disk.
+ * Data is stored at the provided file path.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs a new Storage object.
+     * @param filePath Path of local save file for Duke's task list.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves Duke's TaskList in its current state to a file on the hard disk.
+     * @param tasks TaskList kept by this instance of Duke.
+     */
+    public void store(TaskList tasks) {
+        try {
+            FileWriter writer = new FileWriter(".\\data\\duke.txt");
+            StringBuilder textToWrite = new StringBuilder();
+
+            for (int i = 1; i <= tasks.getNumOfTasks(); i++) {
+                Task t = tasks.retrieve(i);
+                // Insert / separators.
+                String nextEntry =
+                        t.getStringType() + " / " +
```

Comment:
The plus sign should be at the start of the next line to follow coding standard.

Reply:
Thanks. Always forget which way round its supposed to be